### PR TITLE
Documentation changes for monorepo migration

### LINKED ISSUE #6523: Use monorepo in dev docs articles
## 📝 Provide a description of requested docs changes

Guides like [development environment](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/development-environment.html) should be based on monorepo setup, rather than multirepo approach.

Also other docs (or readmes also considered a documentation) might mention multi-repo so this should be addressed in this issue too.

A subtask of #6479

### PR DESCRIPTION
**Pull request by mlewand**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Docs: Documentation changes for monorepo migration. Closes #6523.

---

### Additional information

The "developer environment" guide got heavily simplified, many steps are no longer needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ The main development repository of CKEditor 5 is located at [https://github.com/
 
 ### Development
 
-CKEditor 5 is a modular, multi-package, multi-repository project. It consists of several packages that create the editing framework, based on which the feature packages are implemented.
+CKEditor 5 is a modular, multi-package, monorepo project. It consists of several packages that create the editing framework, based on which the feature packages are implemented.
```
Comment (oleq) [inline README.md:120]:
```suggestion
CKEditor 5 is a modular, multi-package, [monorepo](https://en.wikipedia.org/wiki/Monorepo) project. It consists of several packages that create the editing framework, based on which the feature packages are implemented.
```

File: docs/framework/guides/contributing/development-environment.md [hunk 2]
```diff
@@ -8,13 +8,17 @@ order: 10
 The CKEditor 5 codebase is divided into multiple [npm](http://npmjs.com/) packages, each developed in a separate Git repository. The main package is [`ckeditor5`](https://github.com/ckeditor/ckeditor5) which installs all project dependencies and various development-related resources such as:
```
Comment (Reinmar) [inline docs/framework/guides/contributing/development-environment.md:8]: "each developed in a separate Git repository"

File: docs/framework/guides/contributing/development-environment.md [hunk 3]
```diff
@@ -8,13 +8,17 @@ order: 10
...
-* configuration for [Mr. Git](https://www.npmjs.com/package/mrgit) (a multi-repo management tool) and [Yarn](https://yarnpkg.com/) (a dependency management tool),
+* configuration for [Yarn](https://yarnpkg.com/),
 * translation management tools,
 * documentation generator,
 * and release tools.
 
 You can find all the official packages listed in [CKEditor 5 development repository's README](https://github.com/ckeditor/ckeditor5#packages).
 
+<info-box info>
+After the **19.0.0** release CKEditor 5 moved to the monorepo architecture. If you'd like to work with the older multi-repository release, please refer to [older Developer Environment guide](/docs/ckeditor5/19.0.0/framework/guides/contributing/development-environment.html) for multi-repository oriented instructions.
```
Comment (Reinmar) [inline docs/framework/guides/contributing/development-environment.md:19]: The reader will be reading this from the perspective of CKEditor 5 being a monorepo already. So, what you want to explain here is that "prior to 19.0.0 CKEditor 5 was developed in multi repository architecture".

**Comment by Reinmar** [comment]
There's much more to change than what you proposed @mlewand.

Search for things like:

*   "main repository" which could now be turned into "CKEditor 5 repository",
*   "main CKEditor 5 repository" – same here,
*   the custom builds guide need to be rewritten,
*   same for "installing plugins",
*   the entire README in this package needs to be updated,
*   finally, every package's README and `docs/api/<name>.md` and feature guides (the "Contributing" sections) and most likely more need to be updated,
*   and the list goes on.

It'd be good to have this list split to things that need to be updated right when we'll be migrating and things that can be done within the next iteration. Although, ideally, it'd be good to change all these things at once.

**Comment by Reinmar** [comment]
I can see there are two tickets: #6523 and #6479. The latter is not complete, for sure. I'm afraid that the list will be so long and random that splitting this task to stages may not be possible.

File: README.md (hunk 1 above)
Comment (mlewand) [reply README.md]: Addressed in c2a07ff.

File: docs/framework/guides/contributing/development-environment.md (hunk 2 above)
Comment (mlewand) [reply docs/framework/guides/contributing/development-environment.md]: Addressed in c2a07ff.

File: docs/framework/guides/contributing/development-environment.md (hunk 3 above)
Comment (mlewand) [reply docs/framework/guides/contributing/development-environment.md]: Addressed in c2a07ff.

**Comment by mlewand** [comment]
The PR has been accidentally closed when `monorepo-pending` branch was cleaned up. Now the PR can be merged to the stable branch - as it already has been migrated to monorepo.

**Pull request merged**